moveit_ros/planning_interface/test/move_group_interface_cpp_test.cpp

```diff
@@ -0,0 +1,414 @@
+    {
+      double delta = std::abs(expected[i] - actual[i]);
+      EXPECT_LT(delta, EPSILON) << "joint index: " << i << ", plan: " << expected[i] << ", result: " << actual[i];
+    }
+  }
+
+  void testVectorOfStrings(const std::vector<std::string>& expected, const std::vector<std::string>& actual,
+                           const std::string& name)
+  {
+    SCOPED_TRACE("testVectorOfStrings");
+    ASSERT_EQ(expected.size(), actual.size());
+    for (size_t i = 0; i < actual.size(); ++i)
+      EXPECT_EQ(expected[i], actual[i]) << "(" << name << "[" << i << "])";
+  }
+
+protected:
+  ros::NodeHandle nh_;
+  moveit::planning_interface::MoveGroupInterfacePtr move_group_;
+  moveit::planning_interface::PlanningSceneInterface planning_scene_interface_;
+};
+
+TEST_F(MoveGroupTestFixture, StartingConditionsTest)
+{
+  SCOPED_TRACE("StartingConditionsTest");
+
+  // test that setting the planning time works
+  move_group_->setPlanningTime(PLANNING_TIME_S);
+  EXPECT_EQ(move_group_->getPlanningTime(), PLANNING_TIME_S);
+
+  // test that the world and panda robot is initialized the way we expect by default
+  EXPECT_EQ(move_group_->getNodeHandle().getNamespace(), "/");
+  EXPECT_EQ(move_group_->getEndEffectorLink(), "panda_link8");
+  EXPECT_EQ(move_group_->getEndEffector(), "");
+  EXPECT_EQ(move_group_->getPoseReferenceFrame(), "world");
+  EXPECT_EQ(move_group_->getName(), "panda_arm");
+  EXPECT_EQ(move_group_->getPlanningFrame(), "world");
+  EXPECT_EQ(move_group_->getVariableCount(), std::size_t(7));
+  EXPECT_EQ(move_group_->getDefaultPlannerId(), "");
+  EXPECT_EQ(move_group_->getGoalJointTolerance(), 0.0001);
+  EXPECT_EQ(move_group_->getGoalPositionTolerance(), 0.0001);
+  EXPECT_EQ(move_group_->getGoalOrientationTolerance(), 0.001);
+
+  testVectorOfStrings({ "ready", "extended" }, move_group_->getNamedTargets(), "named_targets");
+  testVectorOfStrings({ "hand", "panda_arm", "panda_arm_hand" }, move_group_->getJointModelGroupNames(),
+                      "joint_model_group_names");
+  testVectorOfStrings({ "panda_joint1", "panda_joint2", "panda_joint3", "panda_joint4", "panda_joint5", "panda_joint6",
+                        "panda_joint7" },
+                      move_group_->getJointNames(), "joint_names");
+  testVectorOfStrings({ "panda_link1", "panda_link2", "panda_link3", "panda_link4", "panda_link5", "panda_link6",
+                        "panda_link7", "panda_link8" },
+                      move_group_->getLinkNames(), "link_names");
+  testVectorOfStrings({ "panda_joint1", "panda_joint2", "panda_joint3", "panda_joint4", "panda_joint5", "panda_joint6",
+                        "panda_joint7" },
+                      move_group_->getActiveJoints(), "active_joints");
+  testVectorOfStrings({ "panda_joint1", "panda_joint2", "panda_joint3", "panda_joint4", "panda_joint5", "panda_joint6",
+                        "panda_joint7", "panda_joint8" },
+                      move_group_->getJoints(), "joints");
```

Comment:
The only aspects of this test the other tests rely on are (1) there are 7 joints and (2) `panda_link0` and `panda_link7` exist. The latter is only needed because of the `OrientationConstraint` test and I would propose to change it to use `getPlanningFrame()` and `getEndEffectorLink()` instead.

Anything else in here that is actually relevant to stay in sync with a tutorial, please refactor it into a different test case `MoveGroupTutorialSupport` and add a comment at the top stating that these tests do not indicate functional breakage, but should never be adjusted without adjusting the respective tutorials as well.

Reply:
I removed the test that checks the default state of the panda arm.  If changes to the panda config break this test they will also break the tutorial unless both are updated as the tests here test the functionality needed for the C++ move_group tutorial.

Reply:
As to using getPlanningFrame() that returns `world` and `getEndEffectorLink()` returns `panda_link8`.  I'm not sure how I would use those to get the strings `panda_link0` and `panda_link7`.  I think it is reasonable that the `OrientationConstraint` will now fail if those link names are changed as it will cause the tutorial to stop working as this is copied from the tutorial

Reply:
I added a comment to explicitly call this out.

Reply:
![panda_frames](https://user-images.githubusercontent.com/680358/78767266-a80b9300-798a-11ea-8d18-fab22232a4cc.png)

Looking at the frames, `world` and `panda_link0` are colocated, `panda_link7` and `panda_link8` are aligned in the z axis.
I don't see why the constraint should fail if you change the frames to the more generic ones, such that the test does not reference implementation details of the model.

Reply:
ah, done